### PR TITLE
fix: props update does not trigger svg reload

### LINKED ISSUE #213: Change preProcessor dose not trigger re-render
### Describe the bug

I wrapped InlineSVG as a component that can change the color in svg, but when I change the incoming color, that is, when the proProcessor changes, re-rendering is not triggered, and the color of svg on the page is still the original.

### Your minimal, reproducible example

none

### Steps to reproduce

```tsx
const XWSvg: FC<XWSvgProps> = ({ color, ...otherProps }) => {
  const preProcessor = useCallback<PreProcessorCallback>(
    (code) => {
      if (color) {
        const svg = new DOMParser().parseFromString(code, 'image/svg+xml');

        const e1 = svg.querySelectorAll('[fill]:not([fill="none"])');
        e1.forEach((e) => e.setAttribute('fill', color));

        const e2 = svg.querySelectorAll('[stroke]:not([stroke="none"])');
        e2.forEach((e) => e.setAttribute('stroke', color));

        return svg.documentElement.outerHTML;
      } else {
        return code;
      }
    },
    [color]
  );

  return <SVG {...otherProps} preProcessor={preProcessor} />;
};

// ---
<XWSvg src={svgIcon} color={isActive ? theme.palette.primary.main : 'green'} />;
```

### Expected behavior

 The svg can update the color when I modify the color field.

### How often does this bug happen?

Every time

### Screenshots or Videos

_No response_

### Platform

_No response_

### react-inlinesvg version

4.0.3

### TypeScript version

4.4.3

### Build tool

webpack

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by BB-fat**
fix #213

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ class ReactInlineSVG extends React.PureComponent<Props, State> {
       }
     }
 
-    if (previousProps.src !== src) {
+    if (previousProps !== this.props) {
```
Comment (gilbarbara) [inline src/index.tsx:82]: This will run every time since `props` is an object and can't be compared with strict equality.

**Review by gilbarbara — CHANGES_REQUESTED** [review]
The change proposed doesn't work

**Comment by gilbarbara** [comment]
As I mentioned in the issue, you can use a `key` in the component to refresh it.
That's the React way.

**Comment by BB-fat** [comment]
Thanks for advice.

**Pull request closed without merging**